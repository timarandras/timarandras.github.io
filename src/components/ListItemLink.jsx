import { ListItem } from '@material-ui/core';
import React from 'react';

export default function ListItemLink(props) {
  return <ListItem button component="a" {...props} />; // eslint-disable-line react/jsx-props-no-spreading
}
