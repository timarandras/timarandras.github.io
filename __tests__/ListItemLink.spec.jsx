import { render } from '@testing-library/react';
import React from 'react';

import ListItemLink from '../src/components/ListItemLink';

describe('ListItemLink', () => {
  test('Matches snapshot', () => {
    const { container } = render(<ListItemLink href="#dummy" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('Renders an <a> with href', () => {
    const { container } = render(<ListItemLink href="#dummy" />);

    expect(container.querySelector('a')).toHaveAttribute('href', '#dummy');
  });
});
