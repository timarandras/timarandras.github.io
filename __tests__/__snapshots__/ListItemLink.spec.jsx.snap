// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemLink Matches snapshot 1`] = `
<a
  aria-disabled="false"
  class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
  href="#dummy"
  tabindex="0"
>
  <span
    class="MuiTouchRipple-root"
  />
</a>
`;
